cuml/random_forest_demo.ipynb

```diff
@@ -0,0 +1,286 @@
+    "\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y,\n",
+    "                                                    test_size = 0.2,\n",
+    "                                                    random_state=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### GPU"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "X_cudf_train = cudf.DataFrame.from_pandas(X_train)\n",
+    "X_cudf_test = cudf.DataFrame.from_pandas(X_test)\n",
+    "\n",
+    "y_cudf_train = cudf.Series(y_train.values)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scikit-learn Model\n",
+    "\n",
+    "### Fit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "sk_model = skrfc(n_estimators=40,\n",
+    "                 max_depth=16,\n",
+    "                 min_samples_split=2,\n",
+    "                 max_features=1.0,\n",
+    "                 random_state=10)\n",
+    "\n",
+    "sk_model.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "sk_predict = sk_model.predict(X_test)\n",
+    "sk_acc = accuracy_score(y_test, sk_predict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## cuML Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "cuml_model = curfc(n_estimators=40,\n",
+    "                   max_depth=16,\n",
+    "                   min_rows_per_node=2,\n",
+    "                   max_features=1.0,\n",
+    "                   seed=10)\n",
+    "\n",
+    "cuml_model.fit(X_cudf_train, y_cudf_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "fil_preds_orig = cuml_model.predict(X_cudf_test)\n",
+    "\n",
+    "fil_acc_orig = accuracy_score(y_test, fil_preds_orig)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pickle the cuML random forest classification model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dump_orig_model = pickle.dumps(cuml_model)\n",
+    "# delete the previous model to ensure there is no leakage of pointers\n",
```

Comment:
Expand a bit to "(this is not strictly necessary but just included here for demo purposes)"